### ISSUE TITLE
DockBuilderCopyDockSpace crashes when windows not in remapping list are docked in a split

### ISSUE THREAD

**Issue by cfillion** [ocornut/imgui#6035]
**Version/Branch of Dear ImGui:**

Version: 1.89.1
Branch: docking

**My Issue:**

Using `DockBuilderCopyDockSpace` to move the contents of a dockspace to another crashes if the dockspace is split, contains a window in the first child (top or left) and nothing in the second child (bottom or right). (It works as expected in the opposite case: something in the second child and nothing in the first.)

**Video**

![Take](https://user-images.githubusercontent.com/4297676/210137364-23d7b3c6-9c52-4d9d-bc81-1450b3c8b09a.gif)

**Standalone, minimal, complete and verifiable example:**
```cpp
#include <imgui/imgui_internal.h>

if(ImGui::Begin("Main")) {
  if(ImGui::Button("Crash Me")) {
    ImVector<const char *> remap;
    ImGui::DockBuilderCopyDockSpace(~0, ~1, &remap);
  }
  ImGui::DockSpace(~0, { 300.f, 300.f });
  ImGui::SameLine();
  ImGui::DockSpace(~1, { 300.f, 300.f });
}
ImGui::End();

if(ImGui::Begin("Dock Me"))
  ImGui::Text("Lorem ipsum");
ImGui::End();
```

**Comment by ocornut** [comment]
Thank you for reporting this.
This happened when the window in the split section is not in the remapping list.
Fixed with 6939676

**Comment by cfillion** [comment]
Thanks, and happy new year! 🎉
